src/spikeinterface/core/generate.py

```diff
@@ -1079,15 +1116,24 @@ def __init__(
         self.num_channels = num_channels
         self.noise_block_size = noise_block_size
         self.noise_levels = noise_levels
+        self.cov_matrix = cov_matrix
         self.dtype = dtype
         self.seed = seed
         self.strategy = strategy
 
         if self.strategy == "tile_pregenerated":
             rng = np.random.default_rng(seed=self.seed)
-            self.noise_block = (
-                rng.standard_normal(size=(self.noise_block_size, self.num_channels), dtype=self.dtype) * noise_levels
-            )
+
+            if self.cov_matrix is None:
```

Comment:
Another option is that the cov is the identity by default and then you don't need this if. But I am not sure if this is less performant. 

Maybe we can do that later.

Reply:
This internally make a np.dot I think so handling None is good no ?

Reply:
I meant that if cov=None then cov=np.eye(number_of_channels) and then you don't have the if.

But I think this version is more memory efficient when cov is None as it avoids the creation of a large matrix.